### PR TITLE
Gen notes Update

### PR DESCRIPTION
**Pull request by mohamedAtoui**
### Description
Change the output of langgraph from NoteType into .md format

### Additional Notes
Some functions are comented which may added later.

### CODE REVIEW

File: student-copilot-frontend/convex/aiAgent/noteAgent.ts [hunk 1]
```diff
@@ -74,217 +36,154 @@ export async function fetchImageLink(query: string): Promise<string> {
...
-      console.error("Unexpected response structure:", response.data);
-      throw new Error("Unexpected response structure: no image link found");
+      console.error('Unexpected response structure:', response.data);
+      throw new Error('Unexpected response structure: no image link found');
     }
   } catch (error) {
     if (axios.isAxiosError(error)) {
-      console.error("Error with Axios:", error.message);
+      console.error('Error with Axios:', error.message);
       if (error.response) {
```
Comment (botirk38) [inline student-copilot-frontend/convex/aiAgent/noteAgent.ts:48]: this function can be simplified, I have a similar fetchImage tool node which is reusable, please look at the utils.ts file

File: student-copilot-frontend/convex/aiAgent/noteAgent.ts [hunk 2]
```diff
@@ -74,217 +36,154 @@ export async function fetchImageLink(query: string): Promise<string> {
...
-  const structuredLlm = llm.withStructuredOutput(NoteBlockSchema);
-  const result = await structuredLlm.invoke(prompt);
 
-  const dataResponse = await NoteBlockSchema.safeParseAsync(result);
+// Function to generate structured lecture notes
+export async function generateTitle(_state: typeof InputAnnotation.State): Promise<{messages:AIMessage}> {
+    
+  
+    const llm = new ChatOpenAI({ model: "gpt-4o-mini-2024-07-18" });
+
```
Comment (botirk38) [inline student-copilot-frontend/convex/aiAgent/noteAgent.ts:84]: Remove these commented out code if not needed.

File: student-copilot-frontend/convex/aiAgent/noteAgent.ts [hunk 3]
```diff
@@ -74,217 +36,154 @@ export async function fetchImageLink(query: string): Promise<string> {
...
+  export const decideIfImageNeeded = async (state: typeof InputAnnotation.State) => {
+  
+    // Invoke the model
+    const llm = new ChatOpenAI({ model: "gpt-4o-mini-2024-07-18"});
+
+    const chain = imageGenerationPrompt.pipe(llm);
+
+    const {  chunk , ImgArr } = state;
+    
+    try {
```
Comment (botirk38) [inline student-copilot-frontend/convex/aiAgent/noteAgent.ts:130]: Same here remove dead code

File: student-copilot-frontend/convex/aiAgent/noteAgent.ts [hunk 4]
```diff
@@ -74,217 +36,154 @@ export async function fetchImageLink(query: string): Promise<string> {
...
-  return fullPlan.trim();
-}
+
 
 // Set up the state graph to control annotation processing flow
 export const graph = new StateGraph({
   input: InputAnnotation,
   output: OutputAnnotation,
 })
-  .addNode("generate_image", generateImageSearchQuery)
```
Comment (botirk38) [inline student-copilot-frontend/convex/aiAgent/noteAgent.ts:288]: we should have a tool node here to generate image for each chunk, by generate I mean searching look at utils.ts file and my flashcard agent as a example

**Review by botirk38 — CHANGES_REQUESTED** [review]

File: student-copilot-frontend/convex/aiAgent/noteAgent.ts (hunk 2 above)
Comment (mohamedAtoui) [reply student-copilot-frontend/convex/aiAgent/noteAgent.ts:84]: I may need it later on,

File: student-copilot-frontend/convex/aiAgent/noteAgent.ts (hunk 3 above)
Comment (mohamedAtoui) [reply student-copilot-frontend/convex/aiAgent/noteAgent.ts:130]: Same may needed in the future.

Comment (botirk38) [reply student-copilot-frontend/convex/aiAgent/noteAgent.ts:130]: doesnt matter, not needed now can delete.

File: student-copilot-frontend/convex/aiAgent/noteAgent.ts (hunk 2 above)
Comment (botirk38) [reply student-copilot-frontend/convex/aiAgent/noteAgent.ts:84]: same as below.

File: student-copilot-frontend/convex/aiAgent/noteAgent.ts (hunk 3 above)
Comment (mohamedAtoui) [reply student-copilot-frontend/convex/aiAgent/noteAgent.ts:130]: what about later on when I need it will take time to write it again :(

Comment (mohamedAtoui) [reply student-copilot-frontend/convex/aiAgent/noteAgent.ts]: I deleted it

File: student-copilot-frontend/convex/aiAgent/noteAgent.ts (hunk 1 above)
Comment (mohamedAtoui) [reply student-copilot-frontend/convex/aiAgent/noteAgent.ts:48]: could u specifiy please cuz in utils.ts  I see only rwo funcitons: exponantionBackoff and getPriceId

Comment (botirk38) [reply student-copilot-frontend/convex/aiAgent/noteAgent.ts:48]: you are not up to date with main bro, git checkout master-> git pull go back to your branch git merge master, then you will be up to date

File: student-copilot-frontend/convex/aiAgent/noteAgent.ts (hunk 3 above)
Comment (botirk38) [reply student-copilot-frontend/convex/aiAgent/noteAgent.ts]: haha, I was gonna say, you can just move into the utils.ts file in aiAgent and export that from there

File: student-copilot-frontend/convex/aiAgent/noteAgent.ts (hunk 1 above)
Comment (mohamedAtoui) [reply student-copilot-frontend/convex/aiAgent/noteAgent.ts:48]: yeah u right

**Review by botirk38 — APPROVED** [review]

File: student-copilot-frontend/convex/noteAction.ts [hunk 5]
```diff
@@ -106,23 +105,19 @@ export const generateNotes = internalAction({
...
 
     // Compile the application state graph with memory checkpointing enabled
-    const appGraph = graph.compile({ checkpointer: memoryManager });
+    const appGraph = graph.compile();
 
     // Configuration for the application with a unique thread ID
-    const executionConfig = { configurable: { thread_id: uuidv4() } };
+    //const executionConfig = { configurable: { thread_id: uuidv4() } };
 
+    let prev_note: string = "";
```
Comment (botirk38) [inline student-copilot-frontend/convex/noteAction.ts:119]: why not use a memory saver?

**Pull request merged**